test/gtest/ucp/ucp_test.cc

```diff
@@ -50,9 +46,24 @@ void ucp_test::cleanup() {
     m_entities.clear();
 }
 
-ucp_test::entity* ucp_test::create_entity() {
+void ucp_test::init() {
```

Comment:
i don't thing we want to force creating 2 entities for every ucp test. for example, might want to test context/worker without endpoint. how about adding another class - ucp_p2p_test - which will create 2 entities, and have sender() and receiver() member functions?